### ISSUE TITLE
TPU compile fail using the new 2.4.0 version

### ISSUE THREAD

**Issue by rwbfd** [tensorflow/tensorflow#45852]
After Colab update its tensorflow version to 2.4.0 recently, I have found that the code which I used to run the example https://colab.research.google.com/drive/1DvLiYBaolddDFok1FttwEkxHNhsRwK_R?usp=sharing has led to very peculiar behavior. In general, the reported bug is 
```
Compilation failure: Expected element type in shape to be arithmetic type for operation subtract; got PRED.
```

I have not been able to pinpoint the root cause of the bug. However, it would seem that all the bug is caused by a substraction operation. When I remove the substraction operation in the code, another substraction issue pops up, albiet in a different location. It is impossible to remove all the substraction in the code. Therefore it would really be helpful if one can pinpoint the problem. 

PS: The code runs perfectly fine under CPU/GPU environment.

**Comment by rwbfd** [comment]
With some experiments, I have found out that the problem is caused by sparsemax layer from tensorflow_addon. When replacing everything with softmax the situation resolves naturally. I will continue to find a way to fix the sparsemax problem.

**Comment by rmothukuru** [comment]
@rwbfd,
Can you please provide access to your Colab File. Thanks!

**Comment by rwbfd** [comment]
@rmothukuru  Thank you for your timely response. This is the address. https://colab.research.google.com/gist/rwbfd/a3a55484cc742bbb674323d3b87b107e/tabnet_tf.ipynb

However, when examing the source code, I find the backward pass for sparsemax activation seems to be missing. In my experiment with PyTorch, this can lead to disasterous bahevior. Is this the case? If so, I will be happy to provide a solution, but first the forward pass must be fixed.

**Comment by frankchn** [comment]
I think this looks like an internal bug, so we might need to release a new version of the TPU runtime to fix this. Will coordinate internally and let you know.

Internal reference: b/173796434

**Comment by frankchn** [comment]
@rwbfd Can you try again next week. We have pushed out a new version of the TPU runtime, but it will take a day or two to propagate to all TPUs.

**Comment by rwbfd** [comment]
@frankchn That would be great! Thanks!

**Comment by rwbfd** [comment]
@frankchn Unfortunately, the same problem still persists.

**Comment by frankchn** [comment]
Ah that's too bad. I'll have to investigate further.

**Comment by sushreebarsa** [comment]
Was able to replicate the issue in TF v2.5,please find the gist [here](https://colab.research.google.com/gist/sushreebarsa/a6cfbc76cab948e5501f98974b6c7e4c/untitled187.ipynb)..Thanks !
